test/vstest.console.UnitTests/Internal/FilePatternParserTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace vstest.console.UnitTests.Internal
+{
+    using Microsoft.Extensions.FileSystemGlobbing;
+    using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Moq;
+    using System.Collections.Generic;
+    using vstest.console.Internal;
+    using vstest.console.Internal.Interfaces;
+
+    [TestClass]
+    public class FilePatternParserTests
+    {
+        private FilePatternParser filePatternParser;
+        private Mock<IMatcherHelper> mockMatcherHelper;
+
+        [TestInitialize]
+        public void TestInit()
+        {
+            this.mockMatcherHelper = new Mock<IMatcherHelper>();
+            this.filePatternParser = new FilePatternParser(this.mockMatcherHelper.Object);
+        }
+
+        [TestMethod]
+        public void FilePatternParserShouldCorrectlySplitPatternAndDirectory()
+        {
+            var patternMatchingResult = new PatternMatchingResult(new List<FilePatternMatch>());
+            this.mockMatcherHelper.Setup(x => x.Execute(It.IsAny<DirectoryInfoWrapper>())).Returns(patternMatchingResult);
+            this.filePatternParser.GetMatchingFiles(@"C:\Users\vanidhi\Desktop\a\c\*bc.dll");
```

Comment:
Make sure we have all the possible pattern types tested. You can use dynamic data attribute to write data driven tests.

Reply:
Added some more patterns and negative case.
There will always be absolute path give to this API since we check before calling this API if the path is relative or absolute. And if the path is relative, we prepend the current working directory to it.